analytics_data_api/v0/models.py

```diff
@@ -263,19 +277,54 @@ def get_users_in_course(
             search.query.must.append(Q('multi_match', query=text_search, fields=['name', 'username', 'email']))
 
         # Sorting
-        order_by_options = (
-            'username', 'email', 'discussions_contributed', 'problems_attempted', 'problems_completed', 'videos_viewed'
-        )
-        sort_order_options = ('asc', 'desc')
-        if order_by not in order_by_options:
-            raise ValueError('order_by value must be one of: {}'.format(', '.join(order_by_options)))
-        if sort_order not in sort_order_options:
-            raise ValueError('sort_order value must be one of: {}'.format(', '.join(sort_order_options)))
         sort_term = order_by if sort_order == 'asc' else '-{}'.format(order_by)
         search = search.sort(sort_term)
 
         return search
 
+    @classmethod
+    def get_course_metadata(cls, course_id):
+        """
+        Returns the number of students belonging to particular cohorts,
+        segments, and enrollment modes within a course.  Returns data in the
+        following format:
+
+        {
+            'cohorts': {
+                <cohort_name>: <learner_count>
+            },
+            'segments': {
+                <segment_name>: <learner_count>
+            },
+            'enrollment_modes': {
+                <enrollment_mode_name>: <learner_count>
+            }
+        }
+        """
+        search = cls.search()
+        search.query = Q('bool', must=[Q('term', course_id=course_id)])
+        search.aggs.bucket('enrollment_modes', 'terms', field='enrollment_mode')
```

Comment:
This is up to you, but it might be a bit more succinct to wrap this in a loop: ```search.aggs.bucket(mode, 'terms', field=mode) for mode in ['enrollment_modes', 'segments', 'cohorts']``` (inspired by inline loops below).